src/HDF5.jl

```diff
@@ -91,6 +91,7 @@ hdf5_type_id(::Type{S}) where {S<:AbstractString} = H5T_C_S1
 
 const BitsType = Union{Bool,Int8,UInt8,Int16,UInt16,Int32,UInt32,Int64,UInt64,Float32,Float64}
 const ScalarType = Union{BitsType,Reference}
+const ScalarOrComplex = Union{ScalarType}
```

Comment:
```suggestion
```

Typo,

unrelated, but I think we have an error  with Complex in general. Right now we have `Complex{<:ScalarTypes}` everywhere, which includes `Reference` which I think needs to be disallowed. In, which case we could add `Complex{<:Real}` to `BitsType` ?